### PR TITLE
Add MissingBracketsCheck

### PR DESCRIPTION
**Pull request by Waishon**
This Check checks for things like
`if(true) doSomething()`
`if(true) { dosomething(); }`

```
if(true)
    doSomething()
```

### CODE REVIEW

File: lib/Checks/MissingBracketsCheck.vala [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program; if not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ * Authored by: Sören Busse
+ */
+
+public class ValaLint.Checks.MissingBracketsCheck : Check {
```
Comment (MarcusWichelmann) [inline lib/Checks/MissingBracketsCheck.vala]: @danrabbit You're probably the guy who knows best how to name these checks. :-)

File: lib/Checks/MissingBracketsCheck.vala [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        }
+
+        /*
+         *   Is there a method call in the same line as the condition
+         *   Example: if(true) { do_something(); }
+         */
+        regex_same_line.match (line, 0, out matchInfo);
+
+        if (matchInfo.matches ()) {
+            mistake_list.add ({ this, line_index, 0, _("Unexpected method call") });
```
Comment (MarcusWichelmann) [inline lib/Checks/MissingBracketsCheck.vala]: @danrabbit Same for these mistake-description-messages. :-)

File: lib/Checks/MissingBracketsCheck.vala [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    private Regex regex_same_line_no_brackets;
+
+    public MissingBracketsCheck () {
+        try {
+            regex_no_brackets = new Regex ("^(if|while|for|foreach) *\\(.*\\) *$");
+            regex_same_line = new Regex ("(if|while|for|foreach) *\\(.*\\) *[{]+ *[a-zA-Z0-9_()]+");
+            regex_same_line_no_brackets = new Regex ("(if|while|for|foreach) *\\(.*\\) *[^{] *[a-zA-Z0-9_()]+");
+        } catch (Error e) {
+            /* Should never happen */
+            error ("There was a parsing error during doing some regex");
```
Comment (MarcusWichelmann) [inline lib/Checks/MissingBracketsCheck.vala]: Maybe print the error message here, too:
`error ("There was a parsing error during doing some regex: %s", e.message);`
And no need for comments like "should never happen". We know that especially these things *do* happen. 😂

File: lib/Checks/MissingBracketsCheck.vala [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+         *   Is there a method call in the same line as the condition
+         *   Example: if(true) { do_something(); }
+         */
+        regex_same_line.match (line, 0, out matchInfo);
+
+        if (matchInfo.matches ()) {
+            mistake_list.add ({ this, line_index, 0, _("Unexpected method call") });
+        }
+
+        return false;
```
Comment (MarcusWichelmann) [inline lib/Checks/MissingBracketsCheck.vala]: Please return true/false according to if one or more mistakes have been found.

**Comment by danrabbit** [comment]
Closing since this seems pretty abandoned

**Pull request closed without merging**